src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py

```diff
@@ -237,6 +238,98 @@ def test_keyvault_key(self, resource_group):
 
 
+class KeyVaultKeyDownloadScenarioTest(ScenarioTest):
+    @ResourceGroupPreparer(name_prefix='cli_test_kv_key_download')
+    def test_keyvault_key_download(self, resource_group):
+        import OpenSSL.crypto
+
+        self.kwargs.update({
+            'kv': self.create_random_name('cli-test-keyvault-', 24),
+            'loc': 'eastus2'
+        })
+        _create_keyvault(self, self.kwargs)
+
+        key_names = [
+            'ec-p256.pem',
+            'ec-p384.pem',
+            'ec-p521.pem',
+            'ec-p256k.pem',
+            'rsa-2048.pem',
+            'rsa-3072.pem',
+            'rsa-4096.pem',
+            'ec-p256-hsm.pem',
+            'ec-p384-hsm.pem',
+            'ec-p521-hsm.pem',
+            'ec-p256k-hsm.pem',
+            'rsa-2048-hsm.pem',
+            'rsa-3072-hsm.pem',
+            'rsa-4096-hsm.pem'
+        ]
+
+        # RSA Keys
+        # * Generate: `openssl genrsa -out rsa-4096.pem 4096`
+        # * Extract public key: `openssl rsa -in rsa-4096.pem -pubout > rsa-4096.pub.pem`
+        # * Public key PEM to DER: `openssl rsa -pubin -inform PEM -in rsa-4096.pub.pem -outform DER -out rsa-4096.pub.der`
+
+        # EC Keys
+        # * Generate: `openssl ecparam -genkey -name secp521r1 -out ec521.key`
+        # * Extract public key (PEM): `openssl ec -in ec-p521.pem -pubout -out ec-p521.pub.pem`
+        # * Extract public key (DER): `openssl ec -in ec-p521.pem -pubout -outform DER -out ec-p521.pub.der`
+
+        for key_name in key_names:
+            var_name = key_name.split('.')[0] + '-file'
+            if 'hsm' in key_name:  # Should be generated
+                if key_name.startswith('rsa'):
+                    rsa_size = key_name.split('-')[1]
+                    self.cmd('keyvault key create --vault-name {kv} -n ' + var_name +
+                             ' -p hsm --kty RSA-HSM --size ' + rsa_size)
+                elif key_name.startswith('ec'):
+                    ec_curve = key_name.split('-')[1]
+                    curve_names = {
+                        'p256': 'P-256',
+                        'p384': 'P-384',
+                        'p521': 'P-521',
+                        'p256k': 'P-256K'
+                    }
+                    self.cmd('keyvault key create --vault-name {kv} -n ' + var_name +
+                             ' -p hsm --kty EC-HSM --curve ' + curve_names[ec_curve])
+            else:  # Should be imported (Have already been generated offline)
+                self.kwargs[var_name] = os.path.join(KEYS_DIR, key_name)
+                self.cmd('keyvault key import --vault-name {kv} -n ' + var_name + ' --pem-file "{' + var_name + '}"')
+
+            der_downloaded_filename = var_name + '.der'
+            pem_downloaded_filename = var_name + '.pem'
+
+            try:
+                self.cmd('keyvault key download --vault-name {kv} -n ' + var_name + ' -f "' + der_downloaded_filename + '" -e DER')
```

Comment:
Let's use [string format](https://docs.python.org/3/library/string.html#format-string-syntax) for readability.

Reply:
@jiasli Refined this.